vaadin-testbench-unit-shared/src/main/java/com/vaadin/flow/component/grid/GridTester.java

```diff
@@ -313,6 +312,139 @@ private Component getRendererItem(int row, Grid.Column<Y> yColumn) {
                 "Target column doesn't have a ComponentRenderer.");
     }
 
+    /**
+     * Get property value for item's LitRenderer in column.
+     *
+     * @param row
+     *            item row
+     * @param columnName
+     *            key/property of column
+     * @param propertyName
+     *            the name of the LitRenderer property
+     * @param propertyClass
+     *            the class of the value of the LitRenderer property
+     * @param <V>
+     *            the type of the LitRenderer property
+     * @return value of renderer's property for the target cell
+     * @throws UnexpectedTypeException
+     *             when the type of the property does not match the
+     * @throws IllegalArgumentException
+     *             when column for property doesn't exist or the target column of
+     *             the cell is not a LitRenderer
+     */
+    public <V> V getLitRendererPropertyValue(int row, String columnName, String propertyName, Class<V> propertyClass) {
+        return getLitRendererPropertyValue(row, getColumn(columnName), propertyName, propertyClass);
+    }
+
+    /**
+     * Get property value for item's LitRenderer in column.
+     *
+     * @param row
+     *            item row
+     * @param column
+     *            column to get
+     * @param propertyName
+     *            the name of the LitRenderer property
+     * @param propertyClass
+     *            the class of the value of the LitRenderer property
+     * @param <V>
+     *            the type of the LitRenderer property
+     * @return value of renderer's property for the target cell
+     * @throws UnexpectedTypeException
+     *             when the type of the property does not match the
+     * @throws IllegalArgumentException
+     *             when column for property doesn't exist or the target column of
+     *             the cell is not a LitRenderer
+     */
+    public <V> V getLitRendererPropertyValue(int row, int column, String propertyName, Class<V> propertyClass) {
+        return getLitRendererPropertyValue(row, getColumns().get(column), propertyName, propertyClass);
+    }
+
+    private <V> V getLitRendererPropertyValue(int row, Grid.Column<Y> column,
+                                              String propertyName, Class<V> propertyClass) {
+        ensureVisible();
+        if (column.getRenderer() instanceof LitRenderer<Y> litRenderer) {
+            var valueProvider = findLitRendererProperty(litRenderer, propertyName);
+            var untypedValue = valueProvider.apply(getRow(row));
+            if (propertyClass.isInstance(untypedValue)) {
+                return propertyClass.cast(untypedValue);
+            } else {
+                throw new UnexpectedTypeException("Target column property value is the wrong type - expected %s, found %s."
```

Comment:
The exception `UnexpectedTypeException` is supposed to be used for constraint validation:
```
Exception raised in the case that the constraint validator resolution cannot determine a suitable validator for a given type.
```
and, moreover, it produces errors `Caused by: java.lang.ClassNotFoundException: jakarta.validation.UnexpectedTypeException` in Quarkus unit tests module.

I suggest to use `IllegalArgumentException` here, as the conflict is between arguments `propertyName` and `propertyClass`.

Reply:
Good catch.